h2o-core/src/main/java/water/fvec/persist/FramePersist.java

```diff
@@ -0,0 +1,189 @@
+package water.fvec.persist;
+
+import jsr166y.CountedCompleter;
+import water.*;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.util.FileUtils;
+
+import java.net.URI;
+
+import static water.fvec.persist.PersistUtils.*;
+
+public class FramePersist {
+
+    static {
+        TypeMap.onIce(FrameMeta.class.getName());
```

Comment:
I guess this is needed because this Iced class is never actually used in DKV? Is that correct? Can you add a small explanation as a comment to the code? Using the construct is usually limited to the very-core classes.

Reply:
as far as I understand any class that I want to be de-serializable by Icer need to be registerd in the TypeMap, if I dont do this I get an error when trying to desrialize because there is no ID for this class in the TypeMap, maybe I am missing something else

Reply:
> if I dont do this I get an error when trying to desrialize because there is no ID for this class in the TypeMap, maybe I am missing something else

The classloader should take care of adding FrameMeta to typemap when the class is loaded. The typemap is then persisted and onIce is called when read back to H2O.

Maybe worth bit of debugging what is going on.

Reply:
My theory is the one above - because the object is never written anywhere, the icer is not needed and typemap not updated. AutoBuffer first starts writing the type map and only after that you actually use the icer and thus update the typemap.

Reply:
strange, this actually only happens in the JUnit test, in normal scenario this is not needed, but the first junit test fails when reading the written file with
`java.lang.ClassNotFoundException:  BAD
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:355)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at water.Weaver.classForName(Weaver.java:61)
	at water.TypeMap.theFreezable(TypeMap.java:249)
	at water.TypeMap.newFreezable(TypeMap.java:232)
	at water.AutoBuffer.get(AutoBuffer.java:770)
	at water.fvec.persist.PersistUtils.read(PersistUtils.java:26)
	at water.fvec.persist.FramePersist.loadFrom(FramePersist.java:114)`
this is not reproducible with python

Reply:
I have dug throught the code and you are exactly right, sir, and I understand what your are saying, I have yet to find a solution.

Reply:
clarification: i can reproduce this with python tests